jetstream/core/orchestrator.py

```diff
@@ -711,34 +758,41 @@ def _generate_thread(self, idx: int):
         except queue.Empty:
           # No new requests, we can't insert, so put back slot.
           my_slots.put(slot, block=False)
+          ThreadDebugLog(
+              thread_name,
+              f"No new ActiveRequest from generate backlog {idx}. "
+              f"Put back the slot.",
+          )
           # If we were blocking and hit the timeout, then retry the loop.
           # Otherwise, we can exit and proceed to generation.
           if block:
             continue
           else:
             break
 
-        # Signal to kill the thread.
```

Comment:
Intended to delete if new_request is None return?

Reply:
Yes, was about to have a separate PR for this one. But yeah, we can delete this in this PR